1.MLP_Binary_Classifier/MLP_utils/utils.py

```diff
@@ -0,0 +1,627 @@
+    confusion_matrix,
+    f1_score,
+    precision_score,
+    recall_score,
+    roc_auc_score,
+    roc_curve,
+)
+from sklearn.model_selection import train_test_split
+
+from .parameters import Parameters
+
+params = Parameters()
+
+
+def data_split(X_vals, y_vals, train, val, test):
+    """
+    This function splits data into
+    training, validation and testing data
+    Note that ratios of splits must be adjusted to achieve true percentage splits
+
+    Example: 80:10:10 : Train:Validation:Testing split
+    after taking 10% of the data for validation remaining is 90% of original data
+
+    Now taking 10% of the remaining 90% does not yield 10% of the original data:
+    We must adjust the ratios by diving the 10% wanted split from the original data
+    by the adjusted 1 - 0.1 = 0.9 ratio of data remaining
+
+    thus finally we have 0.1 / (1 - 0.1) = 0.11111111111111112
+
+    So ~11% of the remaining 90% of data yields 10% of the orginal data. Cheers Mates.
+
+    Parameters:
+        x_vals
+        y_vals
+        train
+        val
+        test
+        params
+    """
+
+    if train + test + val != 1:
+        return
+    # Random seed set for reproducibility
+    seed = 1
+    # split data into train-test
+    X_train, X_val, Y_train, Y_val = train_test_split(
+        X_vals, y_vals, test_size=val, random_state=seed, stratify=y_vals
+    )
+
+    # splitting from the train dataset a second time without repolacement:
+    # we need to adjust the ratio see docstring for more explaination
+    test = test / (1 - val)
+    # split train data into train-validate
+    X_train, X_test, Y_train, Y_test = train_test_split(
+        X_train, Y_train, test_size=test, random_state=seed, stratify=Y_train
+    )
+
+    # reset the index to avoid downstream errors
+    X_train = X_train.reset_index(drop=True)
+    Y_train = Y_train.reset_index(drop=True)
+    X_test = X_test.reset_index(drop=True)
+    Y_test = Y_test.reset_index(drop=True)
+    X_val = X_val.reset_index(drop=True)
+    Y_val = Y_val.reset_index(drop=True)
+
+    return X_train, X_test, X_val, Y_train, Y_test, Y_val
+
+
+# Data class for x and y data
+class Dataset:
+    """
+    A class for formatting data for a data loader
+
+    Attributes:
+    ----------
+    X : Pandas DataFrame
+        the X dimension of data (features)
+    Y : Pandas DataFrame
+        the Y dimension of data (predictor)
+
+    Methods
+    -------
+    __len__:
+        returns the length of the X dimension
+    -------
+    __getitem__:
+        returns a row of the X and Y dimension given an index
+    """
+
+    def __init__(
+        self,
+        X,
+        Y,
+    ):
+        self.X = X
+        self.Y = Y
+
+    def __len__(self):
+        return len(self.X)
+
+    def __getitem__(self, idx):
+        return self.X[idx], self.Y[idx]
+
+
+# based on https://www.kaggle.com/code/ludovicocuoghi/pytorch-pytorch-lightning-w-optuna-opt
+def build_model_custom(trial, in_features):
```

Comment:
There are a handful of hardcoded values in this function, consider abstracting them to be defined as function arguments (can keep their existing defaults!)